src/analyzer.ts

```diff
@@ -95,90 +160,125 @@ export class StreamAnalyzer extends Transform {
   _resolveDependencyAnalysis: (index: DepsIndex) => void;
 
   constructor(config: ProjectConfig) {
-    super({objectMode: true});
-
     this.config = config;
 
     this.loader = new StreamLoader(this);
     this.analyzer = new Analyzer({
       urlLoader: this.loader,
     });
 
-    // Connect the dependencies stream that the analyzer pushes into to the
-    // processing stream which loads each file and attaches the file contents.
-    this._dependenciesStream.pipe(this._dependenciesProcessingStream);
-
     this.allFragmentsToAnalyze = new Set(this.config.allFragments);
     this.analyzeDependencies = new Promise((resolve, _reject) => {
       this._resolveDependencyAnalysis = resolve;
     });
+
+    // Create the vinyl source stream of files to read out of.
+    this._sourcesStream = vinylSrc(this.config.sources, {
+      cwdbase: true,
+      nodir: true,
+    });
+
+    // _sourcesProcessingStream: Pipe the sources stream through...
+    //   1. The resolver stream, to resolve each file loaded via the analyzer
+    //   2. The analyzer stream, to analyze app fragments for dependencies
+    this._sourcesProcessingStream =
+        this._sourcesStream.pipe(new ResolveStream(this))
+            .pipe(new AnalyzeStream(this));
+
+    // _dependenciesProcessingStream: Pipe the dependencies stream through...
+    //   1. The vinyl loading stream, to load file objects from file paths
+    //   1. The resolver stream, to resolve each loaded file for the analyzer
+    this._dependenciesProcessingStream =
+        this._dependenciesStream.pipe(new VinylReaderTransform())
+            .pipe(new ResolveStream(this));
   }
 
   /**
-   * The source dependency stream that Analyzer pushes discovered dependencies
-   * into is connected to the post-processing stream. We want consumers to only
-   * use the post-processed data so that all file objects have contents
-   * loaded by default. This also makes Analyzer easier for us to test.
+   * Return _dependenciesOutputStream, which will contain fully loaded file
+   * objects for each dependency after analysis.
    */
-  get dependencies(): Transform {
+  get dependencies(): NodeJS.ReadableStream {
     return this._dependenciesProcessingStream;
   }
 
-  _transform(file: File, _encoding: string, callback: FileCB): void {
+  /**
+   * Return _sourcesOutputStream, which will contain fully loaded file
+   * objects for each source after analysis.
+   */
+  get sources(): NodeJS.ReadableStream {
+    return this._sourcesProcessingStream;
+  }
+
+  /**
+   * Resolve a loaded file for our analyzer's loader so that analysis is able to
```

Comment:
maybe "so that analysis is able to continue" -> "so that the analyzer can receive a requested file"?